src/pdfxml.cpp

```diff
@@ -709,6 +709,10 @@ wxPdfTable::GetLastRowsOnPage() const
     const double firstBodyRowHeight = iterBodyFirst->second;
     const bool writeHeader = m_headRowLast > m_headRowFirst;
     double headerHeight = 0;
+
+    //m_document->GetHeaderHeight() is header height minus top margin, so we have to consider top margin as well
+    const double topTotalMargin = m_document->GetTopMargin() + m_document->GetHeaderHeight();
```

Comment:
I think this is not enough. We have to take the height of the table header, if it exists. into account, too. This is simple to accomplish, because the table header height is already known, namely as the value of `m_headHeight`. This value can be added unconditionally, because it will be `0`, if the table has no table header:

```C++
const double topTotalMargin = m_document->GetTopMargin() + m_document->GetHeaderHeight() + m_headHeight;
```

BTW, recalculating the table header height is not necessary, because it is already known. That is, the corresponding code can (and should) be removed.